public/app/features/teams/TeamPages.test.tsx

```diff
@@ -1,26 +1,64 @@
-import { shallow } from 'enzyme';
+import { render, screen } from '@testing-library/react';
 import React from 'react';
+import { Provider } from 'react-redux';
 
 import { NavModel, createTheme } from '@grafana/data';
 import { getRouteComponentProps } from 'app/core/navigation/__mocks__/routeProps';
 import { User } from 'app/core/services/context_srv';
+import { configureStore } from 'app/store/configureStore';
 
 import { OrgRole, Team, TeamMember } from '../../types';
 
 import { Props, TeamPages } from './TeamPages';
 import { getMockTeam } from './__mocks__/teamMocks';
 
-jest.mock('@grafana/runtime/src/config', () => ({
-  ...(jest.requireActual('@grafana/runtime/src/config') as unknown as object),
+jest.mock('app/core/components/Select/UserPicker', () => {
+  return { UserPicker: () => null };
+});
+
+jest.mock('app/core/services/context_srv', () => ({
+  contextSrv: {
+    accessControlEnabled: () => false,
+    hasPermissionInMetadata: () => false,
+    hasAccessInMetadata: () => true,
+    user: {},
+  },
+}));
+
+jest.mock('@grafana/runtime', () => ({
+  ...jest.requireActual('@grafana/runtime'),
+  getBackendSrv: () => ({
+    get: jest.fn().mockResolvedValue([{ userId: 1, login: 'Test' }]),
+  }),
   config: {
     licenseInfo: {
       enabledFeatures: { teamsync: true },
+      stateInfo: '',
+      licenseUrl: '',
     },
     featureToggles: { accesscontrol: false },
+    bootData: { navTree: [], user: {} },
+    buildInfo: {
+      edition: 'Open Source',
+    },
+    appSubUrl: '',
   },
+  featureEnabled: () => true,
 }));
 
+// Mock connected child components instead of rendering them
+jest.mock('./TeamSettings', () => {
+  //eslint-disable-next-line
+  return () => <div>Team settings</div>;
+});
+
+jest.mock('./TeamGroupSync', () => {
+  //eslint-disable-next-line
+  return () => <div>Team group sync</div>;
+});
+
```

Comment:
hmmmm i wonder what we should do here. mocking these out feels a bit weird, it's like we're reimplementing enzyme's `shallow` rendering. is there a problem with rendering them normally? 🤔

Reply:
Yeah, this is a special case - each of these components is connected to Redux, so not mocking them would mean that we need to mock separately all their connected props, etc. Since each of these components has their own tests, we're only interested here that this component renders correct page given the params and not that child components behave properly. 

Not sure if this is the best approach, but I think it makes testing easier in this case.

Reply:
cool cool. i think it's fine, it's definitely no worse than before. 😄 but it might be worth thinking about how we can handle this use case better more generally. there must be more places with nested components connected to redux. maybe one for frontend weekly? 🤔

Reply:
This is a bit different case since the component does something internally that is normally handled by React routes, so there's a bit more logic than usual. Normally mocking Redux store is enough, but not in this case.